README.md

```diff
@@ -169,6 +169,107 @@ package doesn't currently support the functionality that you need.
 
 
 
+
+# A Worked Example
+
+
+
+We use a model involving a Gaussian process (GP) -- you don't need to know anything about
+Gaussian processes other than
+1. they are a class of probabilistic model which can be used for regression (amongst other things).
+2. they have some tunable free-parameters that are usually tuned using an iterative
+optimisation algorithm -- typically a variant on gradient descent.
+Ths is representative of a large number of models in ML / statistics / optimisation.
+
+This example can be copy+pasted into a REPL session.
+
+```julia
+# Install some packages.
+using Pkg
+Pkg.add("ParameterHandling")
+Pkg.add("Optim")
+Pkg.add("Zygote")
+Pkg.add("AbstractGPs")
+
+using ParameterHandling # load up this package.
+using Optim # generic optimisation
+using Zygote # algorithmic differentiation
+using AbstractGPs # package containing the models we'll be working with
+
+# Declare a NamedTuple containing an initial guess at parameters.
+raw_initial_params = (
+    k1 = (
+        var=positive(0.9),
+        precision=positive(1.0),
+    ),
+    k2 = (
+        var=positive(0.1),
+        precision=positive(0.3),
+    ),
+    noise_var=positive(0.2),
+)
+
+# Using ParameterHandling.flatten, we can obtain both a Vector{Float64} representation of
+# these parameters, and a mapping from that vector back to the original parameters:
+flat_initial_params, unflatten = ParameterHandling.flatten(raw_initial_params)
+
+# ParameterHandling.value strips out all of the Positive types in initial_params,
+# returning a plain named tuple of named tuples and Float64s.
+initial_params = ParameterHandling.value(raw_initial_params)
+
+# We define unpack to map directly from the flat Vector{Float64} representation to a
+# the NamedTuple representation with all the Positive types removed.
+unpack = ParameterHandling.value ∘ unflatten
```

Comment:
if this is so useful why is it not in the package?
Alt: if it is just Do not Repeat Youself (DRY), 
good examples (and tests) don't follow DRY, they follow some other principle i can't remember the acronym for but it basically mean be really obvious what is happening at each stage

Reply:
It's just that you need to construct `unflatten` before you can construct this. I suppose one could add a function called `build_unpack` or something which just does this composition, but this feels to me like it's similarly clean 🤷